network/Message.cpp

```diff
@@ -743,12 +745,18 @@ void ExtractErrorMessageData(const Message& msg, int& player_id, std::string& pr
     }
 }
 
-void ExtractHostMPGameMessageData(const Message& msg, std::string& host_player_name, std::string& client_version_string) {
+void ExtractHostMPGameMessageData(const Message& msg, std::string& host_player_name, std::string& client_version_string, std::map<std::string, std::string>& dependencies) {
     try {
         std::istringstream is(msg.Text());
         freeorion_xml_iarchive ia(is);
         ia >> BOOST_SERIALIZATION_NVP(host_player_name)
            >> BOOST_SERIALIZATION_NVP(client_version_string);
+        dependencies.clear();
+        try {
```

Comment:
this extra `try catch` won't work, based on recent discoveries. If there is some error when deserializing, such as due to lack of text for `dependencies` in the archive, the exception gets rethrown in the archive destructor, which causes an immediate abort.

You can just use the outer `try` that has the archive in its scope, so that the archive has already been destructed by the time the `catch` block is reached. You can then check if some of the deserialized-into containers are empty if you want to output to the log...

Reply:
Removed unneeded `try catch`.